### DISCUSSION TITLE
Need suggestion when using histogram

### DISCUSSION THREAD

**Discussion by CNicer** [open-telemetry/opentelemetry-cpp#3021] [General]
I use a histogram to count the time consumed by a function. After the function is called, the time consumed by this call is put into a queue, and a thread is used to get the value of the queue and call the histogram record to report it. When the function is called frequently, the number of opentelemetry calls is also high. Are there any optimization suggestions?

**Comment by juanxiuxu** [comment]
@CNicer,

I have the same requirement just like you wrote, could you share the optimization solution here? Thanks a lot

**Comment by Reneg973** [comment]
@CNicer
@juanxiuxu  
Do you have a callstack?
